Repository/ApprovalRepository.php

```diff
@@ -572,24 +565,28 @@ public function findCurrentWeekToApprove(array $users, User $currentUser): int
         $em = $this->getEntityManager()->createQueryBuilder();
         $expr = $em->expr();
         $approvedList = $em
-            ->select('ap')
+            ->select('ap', 'ah', 'ast')
             ->from(Approval::class, 'ap')
             ->join('ap.user', 'u')
             ->join('ap.history', 'ah')
+            ->leftJoin('ah.status', 'ast')
             ->andWhere($expr->in('u.id', ':users'))
             ->setParameter('users', $usersId)
+            ->orderBy('ap.startDate', 'DESC')
             ->getQuery()
             ->getResult();
```

Comment:
Please add this to your v1 branch:

This is called from the MenuSubscriber, which is executed on EVERY page load (to display a badge with a number of open approvals requests, which is a bad idea IMO).
But this makes it super important to make the least possible database calls.

In my test system on the report scree are normally 4 DB calls. 
Activating the plugin suddenly 48 calls are executed (number depends on the numbers of active users).
With this change I was able to reduce this back to 6 calls.

Also, changing the order of calls in the MenuSubscriber makes it possible to only execute these calls only for Teamleads and Admins and NOT on every page load for every user.

Reply:
And: Happy 🎄 🎁